[FSTORE-1090] Concepts & Guides for helper columns and on-demand features (#333)
logicalclocks/logicalclocks.github.io: docs/user_guides/fs/feature_view/training-data.md

X_train, X_val, X_test, y_train, y_val, y_test = feature_view.get_train_validation_test_split(training_dataset_version=1)
```

## Read training data with primary key(s) and event time

> In certain scenarios, for example for time series analysis training data needs to be sorted according to primary key(s) and event time combination.  
> However, they usually are not included in the feature view query as they are not features used for training. To get them pass following attributes  
> `primary_keys=True` and/or `event_time=True`.

Reviewer: ```suggestion
For certain use cases, e.g. time series models, the input data needs to be sorted according to the primary key(s) and event time combination. 
Primary key(s) and event time are not usually included in the feature view query as they are not features used for training.
To retrieve the primary key(s) and/or event time when retrieving training data, you need to set the parameters `primary_keys=True` and/or `event_time=True`.
```

```python
# get a training dataset
X_train, X_test, y_train, y_test = feature_view.get_train_test_split(training_dataset_version=1, 
                                                                     primary_keys=True,
                                                                     event_time=True)
```

!!! note

>     If event time column has the same name in feature groups included in the parent feature view query then the event time of the left most feature group in the query will be returned. If they have different names then
>     all of them will be returned. Join prefix doesn't have any influence on this behaviour.

Reviewer: ```suggestion
    If the event time columns have the same name across all the feature groups included in the feature view, then only the event time of the label feature group (left most feature group in the query) will be returned. If they have different names, then all of them will be returned. The Join prefix does not have any influence on this behaviour.
```
